File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/LocationRepository.java
package com.lattis.ellipse.domain.repository;

import com.google.android.libraries.places.api.model.Place;
import com.lattis.ellipse.domain.model.Bike;
import com.lattis.ellipse.domain.model.Location;
import com.lattis.ellipse.domain.model.LocationSettingsResult;
import com.lattis.ellipse.domain.model.map.Direction;
import com.lattis.ellipse.domain.model.map.PlaceAutocomplete;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;

public interface LocationRepository {

    Observable<Location> getLocationUpdates();

    Observable<List<Bike>> getDistanceForBikes(Location currentLocation, List<Bike> bikes);

    Observable<LocationSettingsResult> getLocationSettings();

    Observable<ArrayList<PlaceAutocomplete>> getPlaces(String constraint);

    Observable<Place> getPlaceBuffer(String placeId);
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/LocationRepository.kt
package com.lattis.domain.repository

import com.google.android.libraries.places.api.model.Place
import com.lattis.domain.models.LocationSettingsResult
import com.lattis.domain.models.map.PlaceAutocomplete
import com.lattis.domain.models.Location
import io.reactivex.rxjava3.core.Observable
import java.util.*

interface LocationRepository {
    fun getLocationUpdates(freshLocationData:Boolean): Observable<Location>
    fun getLocationSettings(): Observable<LocationSettingsResult>
    fun getPlaces(constraint: String): Observable<ArrayList<PlaceAutocomplete>>
    fun getPlaceBuffer(placeId: String): Observable<Place>
}